crates/interpreter/src/opcode/tables.rs

```diff
@@ -0,0 +1,179 @@
+    }
+
+    /// Inserts a boxed instruction into the table at the specified index.
+    #[inline]
+    pub fn insert_boxed(&mut self, opcode: u8, instruction: BoxedInstruction<'a, H>) {
+        *self.get_boxed(opcode) = instruction;
+    }
+
+    /// Replaces a boxed instruction into the table at the specified index, returning the previous
+    /// instruction.
+    #[inline]
+    pub fn replace_boxed(
+        &mut self,
+        opcode: u8,
+        instruction: BoxedInstruction<'a, H>,
+    ) -> BoxedInstruction<'a, H> {
+        core::mem::replace(self.get_boxed(opcode), instruction)
+    }
+
+    /// Replaces a single instruction in the table at the specified index with `f`.
+    #[inline]
+    pub fn update_boxed<F>(&mut self, opcode: u8, f: F)
+    where
+        F: Fn(&DynInstruction<'a, H>, &mut Interpreter, &mut H) + 'a,
+    {
+        update_boxed_instruction(self.get_boxed(opcode), f)
+    }
+
+    /// Replaces every instruction in the table by calling `f`.
+    #[inline]
+    pub fn update_all<F>(&mut self, f: F)
+    where
+        F: Fn(&DynInstruction<'a, H>, &mut Interpreter, &mut H) + Copy + 'a,
+    {
+        // Don't go through `to_boxed` to avoid allocating the plain table twice.
+        match self {
+            Self::Plain(_) => {
+                self.to_boxed_with(|prev| Box::new(move |i, h| f(&prev, i, h)));
+            }
+            Self::Boxed(boxed) => boxed
+                .iter_mut()
+                .for_each(|instruction| update_boxed_instruction(instruction, f)),
+        }
+    }
+}
+
+/// Make instruction table.
+#[inline]
+pub const fn make_instruction_table<H: Host + ?Sized, SPEC: Spec>() -> InstructionTable<H> {
+    // Force const-eval of the table creation, making this function trivial.
+    // TODO: Replace this with a `const {}` block once it is stable.
+    struct ConstTable<H: Host + ?Sized, SPEC: Spec> {
+        _host: core::marker::PhantomData<H>,
+        _spec: core::marker::PhantomData<SPEC>,
+    }
+    impl<H: Host + ?Sized, SPEC: Spec> ConstTable<H, SPEC> {
+        const NEW: InstructionTable<H> = {
+            let mut tables: InstructionTable<H> = [control::unknown; 256];
+            let mut i = 0;
+            while i < 256 {
+                tables[i] = instruction::<H, SPEC>(i as u8);
+                i += 1;
+            }
+            tables
+        };
+    }
+    ConstTable::<H, SPEC>::NEW
+}
+
+/// Make boxed instruction table that calls `f` closure for every instruction.
+#[inline]
+pub fn make_boxed_instruction_table<'a, H, FN>(
+    table: &InstructionTable<H>,
+    mut f: FN,
+) -> BoxedInstructionTable<'a, H>
+where
+    H: Host + ?Sized,
+    FN: FnMut(Instruction<H>) -> BoxedInstruction<'a, H>,
+{
+    core::array::from_fn(|i| f(table[i]))
+}
+
+/// Updates a boxed instruction with a new one.
+#[inline]
+pub fn update_boxed_instruction<'a, H, F>(instruction: &mut BoxedInstruction<'a, H>, f: F)
+where
+    H: Host + ?Sized + 'a,
```

Comment:
Do we needs `'a` here?

Reply:
Yes, I can't really see how to remove it